### PR TITLE
QF-2008 Add provider view for employer requests

### PR DESCRIPTION
**Pull request by karen-hanna**
…s for standards

### CODE REVIEW

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Orchestrators/EmployerRequestOrchestratorTests.cs
Comment (dashton82) [inline src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Orchestrators/EmployerRequestOrchestratorTests.cs:1]: Can probably delete this 😁

**Review by dashton82 — APPROVED** [review]

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/AggregatedEmployerRequestController.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+using SFA.DAS.ProviderRequestApprenticeTraining.Web.Authorization;
+using SFA.DAS.ProviderRequestApprenticeTraining.Web.Infrastructure;
+using SFA.DAS.ProviderRequestApprenticeTraining.Web.Models;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace SFA.DAS.ProviderRequestApprenticeTraining.Web.Controllers
+{
+    [Authorize(Policy = nameof(PolicyNames.HasProviderAccount))]
+    [SetNavigationSection(NavigationSection.EmployerDemand)]
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/AggregatedEmployerRequestController.cs:16]: There will not be a specific navigation menu for RAT - so the navigation section needs to be for home - what ever that is

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Orchestrators/EmployerRequestOrchestratorTests.cs
Comment (karen-hanna) [reply src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Orchestrators/EmployerRequestOrchestratorTests.cs:1]: deleted

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Views/AggregatedEmployerRequest/AggregatedEmployerRequests.cshtml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                                                @aggregatedEmployerRequest.StandardTitle <span class="das-no-wrap">(level @aggregatedEmployerRequest.StandardLevel)</span>
+                                            </h2>
+                                            <p class="govuk-body govuk-!-margin-bottom-1"><b>Sector:</b> @aggregatedEmployerRequest.StandardSector</p>
+                                            <p class="govuk-body govuk-!-margin-bottom-1"><b>Apprentices:</b> @aggregatedEmployerRequest.NumberOfApprentices</p>
+                                            <p class="govuk-body govuk-!-margin-bottom-1"><b>Employers:</b> @aggregatedEmployerRequest.NumberOfEmployers</p>
+                                        </div>
+                                        <div class="govuk-grid-column-one-third govuk-body govuk-!-text-align-right">
+                                            <a class="govuk-link govuk-link--no-visited-state govuk-body" href="">View requests<span class="govuk-visually-hidden">looking for requests for @aggregatedEmployerRequest.StandardTitle</span></a>
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Views/AggregatedEmployerRequest/AggregatedEmployerRequests.cshtml:42]: I must have copied it from somewhere else on the system.  It doesn't state what it should be in the ticket.  Should i just removed the hidden text?

**Review by ChrisJWoodcock — APPROVED** [review]

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Views/AggregatedEmployerRequest/AggregatedEmployerRequests.cshtml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Views/AggregatedEmployerRequest/AggregatedEmployerRequests.cshtml]: The text is not specified I have amended the text so it reads correctly

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/AggregatedEmployerRequestController.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/AggregatedEmployerRequestController.cs]: This has been fixed

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Infrastructure/Api/IProviderRequestApprenticeTrainingOuterApi.cs [hunk 3]
```diff
@@ -6,7 +6,10 @@ namespace SFA.DAS.ProviderRequestApprenticeTraining.Domain.Interfaces
 {
     public interface IProviderRequestApprenticeTrainingOuterApi
     {
-        [Get("/employerrequests/{employerRequestId}")]
+        [Get("/employerrequests/provider/{ukprn}/aggregated")]
+        Task<List<AggregatedEmployerRequestResponse>> GetAggregatedEmployerRequests([Path]long ukprn);
+
+        [Get("/employerrequest/{employerRequestId}")]
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ProviderRequestApprenticeTraining.Infrastructure/Api/IProviderRequestApprenticeTrainingOuterApi.cs:12]: This should be /employerrequests

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Controllers/EmployerRequestControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            _controller?.Dispose();
+        }
+
+        [Test, MoqAutoData]
+        public async Task AggregatedEmployerRequests_ShouldReturnViewWithViewModel(
+            GetAggregatedEmployerRequestsResult aggregatedRequestResult,
+            long ukprn)
+        {
+            // Arrange
+            var claims = new List<Claim>
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Controllers/EmployerRequestControllerTests.cs:43]: Are the claims needed to be mocked?

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Controllers/HomeControllerTests.cs [hunk 5]
```diff
@@ -1,46 +1,57 @@
...
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Options;
+using Moq;
 using NUnit.Framework;
 using SFA.DAS.Provider.Shared.UI.Models;
 using SFA.DAS.ProviderRequestApprenticeTraining.Web.Controllers;
-using SFA.DAS.Testing.AutoFixture;
 
-namespace SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests.Controllers
+namespace SFA.DAS.ProviderRequestApprenticeTraining.Web.Tests.Controllers
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Controllers/HomeControllerTests.cs:9]: This should be UnitTests if the project is UnitTests

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/EmployerRequestController.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        public EmployerRequestController(IEmployerRequestOrchestrator orchestrator, IHttpContextAccessor contextAccessor)
+        {
+            _orchestrator = orchestrator;
+            _contextAccessor = contextAccessor;
+        }
+
+        [HttpGet("{ukprn}/active", Name = ActiveRouteGet)]
+        public async Task<IActionResult> Active(long ukprn)
+        {
+            return View(await _orchestrator.GetActiveEmployerRequestsViewModel(ukprn));
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/EmployerRequestController.cs:33]: I know we discussed this for 2009 but it also doesn't work here so better to just include the 2009 code here as well.

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/HomeController.cs [hunk 7]
```diff
@@ -32,10 +31,10 @@ public IActionResult Index()
             return Redirect(_configuration.DashboardUrl);
         }
 
-        [HttpGet("{ukprn}/start", Name = StartRouteGet)]
+        [HttpGet("start", Name = StartRouteGet)]
         public IActionResult Start()
         {
-            return View();
+            return RedirectToRoute(EmployerRequestController.ActiveRouteGet);
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/HomeController.cs:37]: And here including the ukprn which is in the claim in this method which is just a debug only method

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Models/EmployerRequest/ActiveEmployerRequestsViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+
+namespace SFA.DAS.ProviderRequestApprenticeTraining.Web.Models
+{
+    public class ActiveEmployerRequestsViewModel
+    {
+        public List<ActiveEmployerRequestViewModel> AggregatedEmployerRequests { get; set; }
+
+        public int RequestCount { get { return AggregatedEmployerRequests.Count; } }
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Models/EmployerRequest/ActiveEmployerRequestsViewModel.cs:9]: return AggregatedEmployerRequests?.Count ?? 0

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Infrastructure/Api/IProviderRequestApprenticeTrainingOuterApi.cs (hunk 3 above)
Comment (karen-hanna) [reply src/SFA.DAS.ProviderRequestApprenticeTraining.Infrastructure/Api/IProviderRequestApprenticeTrainingOuterApi.cs:12]: Done

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Controllers/HomeControllerTests.cs (hunk 5 above)
Comment (karen-hanna) [reply src/SFA.DAS.ProviderRequestApprenticeTraining.Web.UnitTests/Controllers/HomeControllerTests.cs:9]: Done

File: src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/HomeController.cs (hunk 7 above)
Comment (karen-hanna) [reply src/SFA.DAS.ProviderRequestApprenticeTraining.Web/Controllers/HomeController.cs:37]: Done

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]
You can't test the Start method in HomeController as the tests run on a release build and the method doesn't exists as it's now Debug

**Review by ChrisJWoodcock — APPROVED** [review]

**Pull request merged**